### DISCUSSION TITLE
[Plugin] Add API to focus on UI Elements

### DISCUSSION THREAD

**Discussion by Crazycolbster** [OpenRCT2/OpenRCT2#21584] [Feature requests]
It’d be helpful to have some way to put focus on a specific UI element, especially textbox widgets. For example, in my message client for an external game, I let the user press ‘ENTER’ to send a message, but I’d like the focus to go right back to the textbox immediately afterwards to send another message.
![image](https://github.com/OpenRCT2/OpenRCT2/assets/43626579/8bb0c046-197a-4dc5-b35b-d7a7e80fe9b6)
I seriously appreciate all the hard work you guys put in!

**Comment by ZehMatt** [comment]
I've modified the title, you are not asking for a hook but rather a function to give a specific widget the active focus.

**Reply by Crazycolbster** [reply]
Oops. My bad. This is what you get when you have an Electrical Engineer do coding. Thanks!

**Comment by autosysops** [comment]
Yes this would be very useful!

**Comment by mrmagic2020** [comment]
Made it possible to focus on textbox widgets in #22213 :)

https://github.com/OpenRCT2/OpenRCT2/assets/104749673/a846b058-f1a3-493d-9fc1-7d512121b202
